File: src/main/java/javabot/operations/LiteralOperation.java
package javabot.operations;

import javabot.BotEvent;
import javabot.Javabot;
import javabot.Message;
import javabot.model.Factoid;
import javabot.dao.FactoidDao;
import org.springframework.beans.factory.annotation.Autowired;

/**
 * @author ricky_clarkson
 */
public class LiteralOperation extends BotOperation {
    @Autowired
    private FactoidDao dao;

    public LiteralOperation(final Javabot bot) {
        super(bot);
    }

    /**
     * @see BotOperation#handleMessage(BotEvent)
     */
    @Override
    public boolean handleMessage(final BotEvent event) {
        final String message = event.getMessage().toLowerCase();
        final String channel = event.getChannel();
        boolean handled = false;
        if (message.startsWith("literal ")) {
            final String key = message.substring("literal ".length());
            final Factoid factoid = dao.getFactoid(key);
            if (factoid != null) {
                getBot().postMessage(new Message(channel, event, factoid.getValue()));
            } else {
                getBot().postMessage(new Message(channel, event, "I have no factoid called \"" + key + "\""));
            }
            handled = true;
        }
        return handled;
    }
}

File: src/main/kotlin/javabot/operations/LiteralOperation.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import jakarta.inject.Inject
import java.util.Locale
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.FactoidDao

class LiteralOperation @Inject constructor(bot: Javabot, adminDao: AdminDao, var dao: FactoidDao) :
    BotOperation(bot, adminDao) {
    override fun handleMessage(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val message = event.value.lowercase(Locale.getDefault())
        if (message.startsWith("literal ")) {
            val key = message.substring("literal ".length)
            val factoid = dao.getFactoid(key)
            responses.add(
                Message(event, if (factoid != null) factoid.value else Sofia.factoidUnknown(key))
            )
        }
        return responses
    }
}
